Source/Scene/parseKTX2.js

```diff
@@ -0,0 +1,225 @@
+import arraySlice from "../Core/arraySlice.js";
+import defined from "../Core/defined.js";
+import PixelFormat from "../Core/PixelFormat.js";
+import RuntimeError from "../Core/RuntimeError.js";
+import VulkanConstants from "../Core/VulkanConstants.js";
+import PixelDatatype from "../Renderer/PixelDatatype.js";
+import { read } from "../ThirdParty/Workers/ktx-parse.modern.js";
+
+var faceOrder = [
+  "positiveX",
+  "negativeX",
+  "positiveY",
+  "negativeY",
+  "positiveZ",
+  "negativeZ",
+];
+
+// Flags
+var colorModelETC1S = 163;
+var colorModelUASTC = 166;
+
+/**
+ * Parses and transcodes KTX2 buffers to an appropriate target format.
+ *
+ * @param {Uint8Array} data Data representing one KTX2 texture.
+ * @param {Object} supportedTargetFormats Target formats available on the current system.
+ * @param {Object} transcoderModule basis_transcoder object.
+ *
+ * @private
+ */
+function parseKTX2(data, supportedTargetFormats, transcoderModule) {
+  if (!defined(supportedTargetFormats)) {
+    throw new RuntimeError(
+      "KTX2 Loader does not know which GPU formats are supported."
+    );
+  }
```

Comment:
I added `Check.typeOf.object` for supportedTargetFormats and transcoderModule; like you mentioned in another comment it'd be impossible for `data` to be undefined because of the early return in `loadKTX2`.